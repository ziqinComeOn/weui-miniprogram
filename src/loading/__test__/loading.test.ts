import path from 'path'
import simulate from 'miniprogram-simulate'

describe('loading', () => {
    const loading = simulate.load(path.resolve(__dirname, '../loading'))

    test('basic', async () => {
        const comp = simulate.render(loading, { type: 'circle' })
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })

    // test('duration', async () => {
    //     const comp = simulate.render(loading, { type: 'dot-gray', duration: 100, show: true })
    //     comp.attach(document.createElement('parent-wrapper'))
    //     await simulate.sleep(0)
    //     expect(comp.toJSON()).toMatchSnapshot()
    //     await simulate.sleep(1000)
    //     comp.setData({ show: false })
    //     expect(comp.toJSON()).toMatchSnapshot()
    // })
})
