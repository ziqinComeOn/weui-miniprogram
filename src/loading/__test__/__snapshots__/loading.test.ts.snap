// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading basic 1`] = `
<main>
  <wx-view
    animation="{{Object {}}}"
    class="wx_loading_view "
    id="wx_loading_view"
    style="display:;"
  >
    <wx-view
      class="weui-loadmore"
    >
      <wx-view
        class="weui-loading"
      />
      <wx-view
        class="weui-loadmore__tips"
      >
        加载中
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;
